drivers/photo-electronic/adpd410x/adpd410x.c

```diff
@@ -62,37 +62,74 @@ int32_t adpd410x_reg_read(struct adpd410x_dev *dev, uint16_t address,
 			  uint16_t *data)
 {
 	int32_t ret;
-	uint8_t buff[] = {0, 0, 0, 0};
+	uint8_t buff[] = {0, 0};
+
+	ret = adpd410x_reg_read_bytes(dev, address, (uint8_t *) buff, 2);
+	if (ret != SUCCESS) {
+		return ret;
+	}
+
+	*data = ((uint16_t)buff[0] << 8) & 0xff00;
+	*data |= buff[1] & 0xff;
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Read a single byte from device register.
+ * @param dev - Device handler.
+ * @param address - Register address.
+ * @param data - Pointer to the register value container.
+ * @param num_bytes - number of bytes to read, max 255
+ * @return SUCCESS in case of success, FAILURE otherwise.
+ */
+int32_t adpd410x_reg_read_bytes(struct adpd410x_dev *dev, uint16_t address,
+				uint8_t *data, uint16_t num_bytes)
+{
+	int32_t ret;
+	uint8_t i;
+	uint8_t *buff;
+
+	if (num_bytes > 255)
```

Comment:
Is there a reason for this cap or is it arbitrary?

Reply:
For I2C, reads are at most 255 bytes. SPI doesn't seem to have this limit, so I'll move the cap to the ADPD4101 section.